ElementX/Sources/Screens/RoomScreen/View/SwipeRightActionMenu.swift

```diff
@@ -0,0 +1,173 @@
+//
+// Copyright 2023 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import SwiftUI
+
+struct SwipeRightActionMenu: ViewModifier {
+    private let actionThreshold = 50.0
+    private let translationThreshold = 100.0
```

Comment:
I think we should clearly indicate this is the final offset applied to the view and not the translation in the swipe gesture.
```suggestion
    private let xOffsetThreshold = 100.0
```